providers/user.rb

```diff
@@ -0,0 +1,87 @@
+use_inline_resources if defined?(use_inline_resources)
+
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+action :upsert do
```

Comment:
@sunggun-yu 
Sure we can, but I've called it upsert because it is upsert. If user already exists it will be updated otherwise it is created.